lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -86,7 +86,12 @@ ImageDecoderImpeller::ImageDecoderImpeller(
 ImageDecoderImpeller::~ImageDecoderImpeller() = default;
 
 static SkColorType ChooseCompatibleColorType(SkColorType type) {
-  return kRGBA_8888_SkColorType;
+  switch (type) {
+    case kRGBA_F32_SkColorType:
+      return kRGBA_F16_SkColorType;
```

Comment:
Why not
```suggestion
    case kRGBA_F32_SkColorType:
    case kRGBA_F16_SkColorType:
      return kRGBA_F16_SkColorType;
```

Reply:
There isn't a way to send in kRGBA_F16_SkColorType directly.  If we decide to implement that it will probably be better to do that in its own pr with its own tests.